torch/_dynamo/guards.py

```diff
@@ -445,6 +445,18 @@ def DICT_KEYS(self, guard):
     def WEAKREF_ALIVE(self, guard):
         self._produce_guard_code(guard, [f"{self.arg_ref(guard)} is not None"])
 
+    def NN_MODULE_BUFFER_NAMES(self, guard):
+        ref = self.arg_ref(guard)
+        value = self.get(guard.name)
+        t = type(value)
+        keys = {k for k, v in value.named_buffers()}
```

Comment:
Don't get every buffer - this is reimplementing the nn_module guards and we don't want that.

Instead, guard only on the buffer accessed in your setattr.

Reply:
Also, make sure the guard is either a top level guard added manually under guards.py, or propagated on the buffer object with a guard_source that is not nn module. If we see an nn module guard source, we will ignore it.